app/src/main/java/com/appsubaruod/comicviewer/model/file/FileOperator.java

```diff
@@ -50,12 +56,12 @@ public void registerCallback(OnFileCopy mCallback) {
         mCallbackSet.add(mCallback);
     }
 
-    public void unRegisterCallback(OnFileCopy mCallback) {
+    public void unregisterCallback(OnFileCopy mCallback) {
         mCallbackSet.remove(mCallback);
     }
 
     public void unpackZip(Uri uri) throws IllegalFormatException {
-
+        String dirnameToCreate = getStoringDirString(uri);
```

Comment:
FileOperator does not responsible to the place where the uri should be extracted.
Path information should be specified from FileOrganizer.